termsaverlib/screen/img2ascii.py

```diff
@@ -0,0 +1,188 @@
+
+    $ %(app_name)s %(screen)s -p /path/to/my/image.jpg
+    This will trigger the screensaver to read just one image, refreshing at
+    the preset interval.
+
+    $ %(app_name)s %(screen)s -p /path/to/my/images -c -i
+    This will trigger the screensaver to read all files in the path selected
+    using a contrast characterset and inverted colors.
+
+    $ %(app_name)s %(screen)s -p /path/to/my/images -s "1234567890_."
+    This will trigger the screensaver to read all files in the path selected
+    rendering them using only the characters provided.
+    
+""") % {
+        'screen': self.name,
+        'app_name': constants.App.NAME,
+        'default_delay': constants.Settings.CHAR_DELAY_SECONDS,
+    })
+
+
+    def _message_no_path(self):
+        """
+        The specific helper message in case there is no path informed in the
+        command-line arguments.
+        """
+        return _("""
+You just need to provide the path to the location from where %(app_title)s will
+read and display on screen.
+
+If you do not have any code in your local machine, just get some interesting
+project from the Internet, such as Django (http://www.djangoproject.com):
+
+    If you have access to git, you may download it at:
+        git clone https://github.com/django/django.git
+
+    Or, just download the zipped source and unpack it on your local machine:
+        https://www.djangoproject.com/download/
+""") % {
+       'app_title': constants.App.TITLE,
+    }
+
+
+    def __init__(self):
+        """
+        Creates a new instance of this class (used by termsaver script)
+
+        From its base classes, the functionality provided here bases on the
+        settings defined below:
+
+            * clean up each cycle: True
+              this will force the screen to be cleaned (cleared) before
+              each new cycle is displayed
+
+            * clean up each file: True
+              this will force the screen to be cleaned (cleared) before
+              each new file is displayed
+        """
+        ImageReaderBase.__init__(self,
+            "img2ascii",
+            _("displays images in typing animation"),
+            cli_opts={
+                'opts': 'hicd:p:w:s:',
+                'long_opts': ['help', 'invert', 'path=', 'wide=', 'contrast', 'set='],
+            })
+        self.cleanup_per_cycle = True
+        self.cleanup_per_file = True
+        self.options = {}
+
+    def _parse_args(self, prepared_args):
+        for o, a in prepared_args[0]:  # optlist, args
+            if o in ("-h", "--help"):
+                self.usage()
+                self.screen_exit()
+            elif o in ("-i", "--invert"):
+                self.options['invert'] = True
+            elif o in ("-w", "--wide"):
+                try:
+                    # makes sure this is a valid integer
+                    self.options['wide'] = int(a)
+                except:
+                    raise exception.InvalidOptionException("wide")
```

Comment:
All exception.XXX raises error in Python, because the library is not imported.
As an example, just try to run: 
`termsaver img2ascii -v`

Reply:
Yep, not sure how that slipped by. Completely missing the termsaverlib.exception import.